front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +59,44 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points as NonNullable<
+          PathProperties['operational_points']
+        >;
+        // Check if there are vias added by map click and insert them in the operational points
+        if (trainScheduleUsedForProjection.path.some((step) => 'track' in step)) {
```

Comment:
Do we really need this `if`? The `forEach` already has a check to ignore steps which aren't track offsets no?

Reply:
That was to avoid looping through all `trainSchedule.path` with the forEach if there isn't at least one step with `track` but you right it's probably overkill considering the size of the array.

Reply:
`some()` also loops on all elements, so it doesn't actually result in any performance benefit. (And yeah micro-optimizations blah blah blah)